### PR TITLE
fix change branches position while adding nodes

### PR DESCRIPTION
**Pull request by trikimahdi**

### CODE REVIEW

File: src/components/graph/network-modification-tree-model.js [hunk 1]
```diff
@@ -19,9 +29,50 @@ export default class NetworkModificationTreeModel {
...
-    addChild(newNode, parentId, insertMode) {
-        // Add node
-        this.treeNodes.push(convertNodetoReactFlowModelNode(newNode, parentId));
+    addChild(newNode, parentId, insertMode, referenceNodeId) {
+        // we have to keep a precise order of nodes in the array to avoid gettings children
+        // nodes before their parents when building graph in dagre library which have uncontrolled results
+        // We also need to do this to keep a correct order when inserting nodes and not loose the user.
+        const referenceNodeIndex = referenceNodeId
+            ? this.treeNodes.findIndex((node) => node.id === referenceNodeId)
+            : null;
```
Comment (archendel) [inline src/components/graph/network-modification-tree-model.js:38]: what will happen if referenceNodeIndex is null ?

File: src/components/graph/network-modification-tree-model.js [hunk 2]
```diff
@@ -19,9 +29,50 @@ export default class NetworkModificationTreeModel {
...
+        switch (insertMode) {
+            case 'BEFORE': {
```
Comment (archendel) [inline src/components/graph/network-modification-tree-model.js:40]: create enum for "BEFORE", "AFTER", "CHILD"

File: src/components/graph/network-modification-tree-model.js [hunk 3]
```diff
@@ -8,6 +8,16 @@
...
 
+// Function to count children nodes for a given parentId recursively in an array of nodes.
+function countNodes(nodes, parentId) {
+    return nodes.reduce((acc, n) => {
+        if (n.data.parentNodeUuid === parentId) {
+            acc += 1 + countNodes(nodes, n.id); // this node + its children
+        }
+        return acc;
+    }, 0);
+}
```
Comment (archendel) [inline src/components/graph/network-modification-tree-model.js:19]: sounds unoptimal to run "nodes.reduce" recursively on all nodes

there might be a better approach

File: src/components/network-modification-tree-pane.js [hunk 4]
```diff
@@ -152,6 +152,8 @@ export const NetworkModificationTreePane = ({
     }, [broadcastChannel]);
 
     const [activeNode, setActiveNode] = useState(null);
+    const activeNodeRef = useRef();
```
Comment (SlimaneAmar) [inline src/components/network-modification-tree-pane.js:155]: const activeNodeRef = useRef(activeNode);

**Review by SlimaneAmar — APPROVED** [review]

File: src/components/network-modification-tree-pane.js (hunk 4 above)
Comment (archendel) [reply src/components/network-modification-tree-pane.js:155]: "activeNodeRef" to remove, not needed anymore

**Review by archendel — CHANGES_REQUESTED** [review]

File: src/components/graph/network-modification-tree-model.js (hunk 1 above)
Comment (trikimahdi) [reply src/components/graph/network-modification-tree-model.js]: condition removed

File: src/components/network-modification-tree-pane.js (hunk 4 above)
Comment (trikimahdi) [reply src/components/network-modification-tree-pane.js]: "activeNodeRef" to removed

File: src/components/graph/network-modification-tree-model.js (hunk 2 above)
Comment (trikimahdi) [reply src/components/graph/network-modification-tree-model.js]: done

File: src/components/graph/network-modification-tree-model.js (hunk 3 above)
Comment (trikimahdi) [reply src/components/graph/network-modification-tree-model.js]: I added a comment for this

File: src/components/utils/node-insertion-methods.ts
Comment (archendel) [inline src/components/utils/node-insertion-methods.ts:1]: rename to NodeInsertionMode for homogenisation ? (notification is named "insertMode")

**Review by archendel — APPROVED** [review]

**Pull request merged**